IOPool/Streamer/src/StreamerInputFile.cc

```diff
@@ -171,7 +210,7 @@ namespace edm {
           << "Failed reading streamer file, init header size from data too small\n";
     }
 
-    startMsg_ = std::make_shared<InitMsgView>(&headerBuf_[0]);  // propagate_const<T> has no reset() function
+    startMsg_ = std::make_shared<InitMsgView>(headerPtr);  // propagate_const<T> has no reset() function
```

Comment:
Isn't there a problem here (and maybe there was a problem with this code before this change as well)? `startMsg_` stays around until the next file is read. But it is internally holding a pointer to what `headerPtr` was addressing at this point. However, with this change you are using `noCopy` so `headerPtr` in this routine points to something in `tmpBuffer_` which will change so when `startMsg_` is read at next file boundary, the memory it reads will not be the same as when it was called here.

Reply:
In StreamerInputFile it's only used to get currRun_ and currProto_, they are fetched when the message is new (after `readStartMessage`), so that is ok. There is also a check if smart pointer of InitMsgView isn't null in `openNextFile`, but that doesn't rely on the tmpBuffer_.
And in `StreamerFileReader.cc` the header is always accessed and deserialized after opening a new fileand not kept around for later.
So I think it's ok, but I will have another look.

Reply:
After anothe look, it is safe. Also, a unit test job now opens two files larger than the prefetch buffer successively, so a corrupted INI would be detected.